src/snowflake/cli/api/project/schemas/entities/application_package_entity_model.py

```diff
@@ -75,3 +75,20 @@ def append_test_resource_suffix_to_identifier(
         if isinstance(input_value, Identifier):
             return input_value.model_copy(update=dict(name=with_suffix))
         return with_suffix
+
+    @field_validator("artifacts")
+    @classmethod
+    def transform_artifacts(
+        cls, orig_artifacts: List[Union[PathMapping, str]]
+    ) -> List[PathMapping]:
```

Comment:
This will be needed by Snowpark + Streamlit entities as well, let's chat with CLI on Tuesday to figure out how to combine efforts here